src/@ionic-native/plugins/fcm/index.ts

```diff
@@ -0,0 +1,171 @@
+import { Plugin, IonicNativePlugin, Cordova } from '@ionic-native/core';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+
+export interface NotificationData {
+
+  /**
+   * Determines whether the notification was pressed or not
+   */
+
+  wasTapped: boolean;
+
+  /**
+   * Notification data hash item
+   */
+
+  [name: string]: any;
+
+}
+
+/**
+ * @name FCM
+ * @description
+ * Provides basic functionality for Firebase Cloud Messaging
+ *
+ * @usage
+ * ```typescript
+ * import { FCM } from 'ionic-native';
+ *
+ * constructor(private fcm: FCM) {}
+ * 
+ * ...
+ * 
+ * fcm.subscribeToTopic('marketing');
+ * 
+ * fcm.getToken().then(token=>{
+ *   backend.registerToken(token);
+ * })
+ * 
+ * fcm.onNotification().subscribe(data=>{
+ *   if(data.wasPressed){
+ *     console.log("Received in background");
+ *   } else {
+ *     console.log("Received in foreground");
+ *   };
+ * })
+ * 
+ * fcm.onTokenRefresh().subscribe(token=>{
+ *   backend.registerToken(token);
+ * })
+ * 
+ * fcm.unsubscribeToTopic('marketing');
+ * 
+ * ```
+ * @interfaces
+ * NotificationData
+ */
+@Plugin({
+  pluginName: 'FCM',
+  plugin: 'cordova-plugin-fcm',
+  pluginRef: 'FCMPlugin',
+  repo: 'https://github.com/fechanique/cordova-plugin-fcm',
+  platforms: ['iOS', 'Android']
+})
+@Injectable()
+export class FCM extends IonicNativePlugin {
+
+  /**
+   * Get's device's current registration id
+   * 
+   * ```typescript
```

Comment:
Keep all usage examples in the main block above the `@Plugin` decorator. Do not add any usage documentation above the method declaration.